v2/contribute/decisions/session/0022-handshake-after-jwt-access-token.mdx

```diff
@@ -0,0 +1,39 @@
+---
+id: "0022"
+title: After the access token becomes a full JWT, we can drop the backend-core handshake
+hide_title: true
+---
+
+import DecisionHeader from "/src/components/decisionLogs/DecisionHeader"
+import ArgumentPro from "/src/components/decisionLogs/ArgumentPro"
+import ArgumentCon from "/src/components/decisionLogs/ArgumentCon"
+import ArgumentNeut from "/src/components/decisionLogs/ArgumentNeut"
+
+# After the access token becomes a full JWT, we can drop the backend-core handshake
+
+<DecisionHeader status="proposed" lastUpdate="2022-11-28" created="2022-11-28" deciders={["rishabhpoddar", "porcellus"]} proposedBy={["porcellus"]} />
+
+## Context and Problem Statement
+
+The backend-core handshake is mainly used to fetch a few config values and the list of valid signing public keys we can use to locally validate access tokens. The fetched config values are:
+- Anti-csrf mode
+- Access token blacklisting enabled
+- Access token validity
+- Refresh token validity
+
+## Considered Options
+
+* Keep the handshake
+* Remove handshake
+
+## Decision Outcome
+
+We can drop the handshake:
+- Anti-csrf mode can be defined on the backend
+- Access token blacklisting will be moved to the backend
+- Validities are already sent in the create and refresh core API responses
+- The core JWKs endpoint can return the necessary public keys
```

Comment:
Would dropping handshake also work for if JWT key rotation is enabled such that:
- Only new keys are added periodically
- New keys are added periodically and older keys are auto removed.

Is there any edge case you can think of in which the SDK will fail to query the JWKs endpoint? Or maybe the user does not want the endpoint to exist (which is possible if they disable that API), so then the SDK will have to use the handshake way as a fallback?

Reply:
- Looking at how JWK client libs seem to handle caching (i.e.: https://github.com/auth0/node-jwks-rsa/blob/master/EXAMPLES.md#caching), dropping the handshake would work in both cases.
- This should still be queried from the core, so the backend failing to communicate with it would be a problem. I don't think we should disable this endpoint there. This could have the same auth checks as other core endpoints, we could make that part configurable, but I don't see a reason to disable this endpoint as it only returns public information.

Reply:
Making it so that user's can't disable this endpoint is going against our architecture. Which i don't like. That being said, the point of just querying the core instead makes sense, however, the format in which the core returns the public keys is very different and not directly consumable by jwks libraries. So we might need to change that maybe?

Reply:
I was talking about a (possibly new) core jwks endpoint, that conforms to the standard. I'm unsure how much we differ from the standard besides the location. 
The core should still be the one signing the access tokens, so we have to query the core to fetch the public keys.
The only other option is signing in the backend SDKs (but I don't think we should). Then we wouldn't need to retrieve any keys to verify the tokens because it's right there in the backend SDK memory.

Reply:
Making the core URL conform with the standard makes sense.